readium/navigator/src/main/java/org/readium/r2/navigator/epub/EpubNavigatorFragment.kt

```diff
@@ -271,62 +262,50 @@ class EpubNavigatorFragment private constructor(
+                    resourcesSingle.add(PageResource.EpubFxl(leftLink = link, leftUrl = url))
 
                     // add first page to the right,
                     if (index == 0) {
-                        resourcesDouble.add(PageResource.EpubFxl("", url))
+                        resourcesDouble.add(PageResource.EpubFxl(rightLink = link, rightUrl = url))
                     } else {
                         // add double pages, left & right
-                        if (doublePageLeft == "") {
-                            doublePageLeft = url
+                        if (doublePageLeft == null) {
+                            doublePageLeft = link
                         } else {
-                            doublePageRight = url
+                            doublePageRight = link
                             resourcesDouble.add(
                                 PageResource.EpubFxl(
-                                    doublePageLeft,
-                                    doublePageRight
+                                    leftLink = doublePageLeft,
+                                    leftUrl = viewModel.urlTo(doublePageLeft),
+                                    rightLink = doublePageRight,
+                                    rightUrl = viewModel.urlTo(doublePageRight),
                                 )
                             )
-                            doublePageLeft = ""
+                            doublePageLeft = null
                         }
                     }
                 }
                 // add last page if there is only a left page remaining
-                if (doublePageLeft != "") {
-                    resourcesDouble.add(PageResource.EpubFxl(doublePageLeft, ""))
+                if (doublePageLeft != null) {
+                    resourcesDouble.add(PageResource.EpubFxl(leftLink = doublePageLeft, leftUrl = viewModel.urlTo(doublePageLeft)))
                 }
 
                 this.resourcesSingle = resourcesSingle
                 this.resourcesDouble = resourcesDouble
 
                 resourcePager.type = Publication.TYPE.FXL
-                adapter = when (preferences.getInt(COLUMN_COUNT_REF, 0)) {
-                    1 -> {
-                        R2PagerAdapter(childFragmentManager, resourcesSingle)
-                    }
-                    2 -> {
-                        R2PagerAdapter(childFragmentManager, resourcesDouble)
-                    }
-                    else -> {
-                        // TODO based on device
-                        // TODO decide if 1 page or 2 page
-                        R2PagerAdapter(childFragmentManager, resourcesSingle)
-                    }
-                }
             }
         }
-        adapter.listener = PagerAdapterListener()
 
-        resourcePager.adapter = adapter
-        resourcePager.direction = publication.metadata.effectiveReadingProgression
+        resetAdapter()
```

Comment:
The adapter can now be reset when changing settings that require a relayout (e.g. reading progression).